.storybook/preview.tsx

```diff
@@ -20,36 +27,82 @@ const ThemeBlock = styled.div<{
 `;
 
 const SideBySideContainer = styled.div`
+  width: 100%;
+  height: 100%;
   display: flex;
   justify-content: space-around;
-  width: 100%;
   > * {
     flex: 1;
   }
 `;
 
+type StorybookTheme = 'day' | 'night' | 'side-by-side';
+
 const withTheme: Decorator = (Story, context) => {
-  const theme = context.parameters.theme || context.globals.theme;
+  const [globals, updateGlobals] = useGlobals();
+
+  const storybookTheme: StorybookTheme =
+    context.parameters.theme || globals.theme;
+
+  const { aminoTheme, setAminoTheme } = useAminoTheme({ root: true });
+  const previousStorybookTheme = usePrevious(storybookTheme);
+
+  const sameTheme = storybookTheme === aminoTheme;
+  const storybookChanged = previousStorybookTheme !== storybookTheme;
+
+  // Sync amino theme with storybook
+  useEffect(() => {
+    if (previousStorybookTheme !== storybookTheme) {
+      if (!sameTheme && storybookTheme !== 'side-by-side') {
```

Comment:
What do you think about avoiding nested if statements?
```suggestion
    if (previousStorybookTheme !== storybookTheme && !sameTheme && storybookTheme !== 'side-by-side') {
```

Reply:
Leftover from my brain trying to figure out the logic to sync these states.. 🤦